### PR TITLE
custom form

### PR DESCRIPTION
**Pull request by do-dyco**

### CODE REVIEW

**Comment by yughoz** [comment]
https://github.com/user-attachments/assets/9ee3b66c-7865-4e87-8131-43908c17e3b5

custom logo size and correction

**Comment by atmonshi** [comment]
thank you @do-dyco and @yughoz 
there is two PRs with similar changes, can we agree on one and close the other :)

**Comment by yughoz** [comment]
> thank you @do-dyco and @yughoz there is two PRs with similar changes, can we agree on one and close the other :)

Let me know if you need any adjustments!

File: src/Facades/Qr.php [hunk 1]
```diff
@@ -64,6 +65,18 @@ public static function getFormSchema(
                         ->statePath($optionsStatePath)
                         ->schema([
                             Hidden::make('type')->default('png'),
+                            Select::make('correction')
```
Comment (atmonshi) [inline src/Facades/Qr.php:76]: since `correction` is a new option, we need to add it to the `getDefaultOptions` too

File: src/Facades/Qr.php [hunk 2]
```diff
@@ -84,6 +97,7 @@ public static function getFormSchema(
                                     '9' => '9',
                                 ]),
 
+
```
Comment (atmonshi) [inline src/Facades/Qr.php]: duplicated empty line :)

File: src/Facades/Qr.php [hunk 3]
```diff
@@ -146,7 +160,7 @@ public static function getFormSchema(
                                 ])
                                 ->columnSpan(['sm' => 2])
                                 ->columns(['sm' => 2])
-                                ->visible(fn (Get $get) => $get('hasGradient')),
+                                ->visible(fn(Get $get) => $get('hasGradient')),
```
Comment (atmonshi) [inline src/Facades/Qr.php]: can you revert the code formatting

File: src/Facades/Qr.php [hunk 4]
```diff
@@ -182,21 +196,38 @@ public static function getFormSchema(
                                 ])
                                 ->columnSpan(['sm' => 2])
                                 ->columns(['sm' => 2])
-                                ->visible(fn (Get $get) => $get('hasEyeColor')),
+                                ->visible(fn(Get $get) => $get('hasEyeColor')),
```
Comment (atmonshi) [inline src/Facades/Qr.php]: can you revert the code formatting  and empty line

File: src/Facades/Qr.php [hunk 5]
```diff
@@ -182,21 +196,38 @@ public static function getFormSchema(
...
+
+
 
                             FileUpload::make('logo')
                                 ->live()
                                 ->imageEditor()
                                 ->columnSpanFull()
                                 ->image(),
+
+                            Select::make('percentage')
```
Comment (atmonshi) [inline src/Facades/Qr.php:216]: also add `percentage` to `getDefaultOptions`

File: src/Facades/Qr.php [hunk 6]
```diff
@@ -182,21 +196,38 @@ public static function getFormSchema(
...
+                                ]),
                         ]),
 
                     Placeholder::make('preview')
                         ->label(__('Preview'))
                         ->columns(['sm' => 2])
                         ->columnSpan(['sm' => 2, 'lg' => 1])
                         ->key('preview_placeholder')
-                        ->content(fn (Get $get) => Qr::render(
+                        ->content(fn(Get $get) => Qr::render(
```
Comment (atmonshi) [inline src/Facades/Qr.php]: code formatting

File: src/Facades/Qr.php [hunk 7]
```diff
@@ -210,6 +241,8 @@ public static function getFormSchema(
     public static function output(?string $data = null, ?array $options = null): HtmlString
     {
         $maker = new Generator;
+        // $maker->errorCorrection('L');
```
Comment (atmonshi) [inline src/Facades/Qr.php]: if we added the `percentage` to the default array we dont need this default variable

File: src/Facades/Qr.php [hunk 8]
```diff
@@ -267,8 +309,22 @@ public static function output(?string $data = null, ?array $options = null): Htm
             reset($options['logo']);
             $logo = current($options['logo']);
 
-            if (filled($logo->getPathName())) {
-                $maker = $maker->merge($logo->getPathName(), .4, true);
+
+            if ($logo instanceof \Illuminate\Http\UploadedFile && filled($logo->getPathName())) {
```
Comment (atmonshi) [inline src/Facades/Qr.php]: I am not following this part and why we need it?
if the image loaded from a storage path I dont think we need to upload it again, right?

**Review by atmonshi — CHANGES_REQUESTED** [review]
thank you so much for the PR, these are small code refactoring

**Comment by aSmodeusW** [comment]
@yughoz, @atmonshi  can you tell me how you overcame issue of uploading qr code? i can upload it but when i want to view that image of an qr code in an edit,it says Call to a member function getPathName() on string

**Comment by yughoz** [comment]
> @yughoz, @atmonshi can you tell me how you overcame issue of uploading qr code? i can upload it but when i want to view that image of an qr code in an edit,it says Call to a member function getPathName() on string

I have updated in my git
just waiting for the PR to be approved

**Comment by atmonshi** [comment]
fix #39

**Pull request merged**